symfony/stimulus-bundle/2.20/assets/controllers/csrf_protection_controller.js

```diff
@@ -1,61 +1,79 @@
-var nameCheck = /^[-_a-zA-Z0-9]{4,22}$/;
-var tokenCheck = /^[-_/+a-zA-Z0-9]{24,}$/;
+const nameCheck = /^[-_a-zA-Z0-9]{4,22}$/;
+const tokenCheck = /^[-_\/+a-zA-Z0-9]{24,}$/;
 
 // Generate and double-submit a CSRF token in a form field and a cookie, as defined by Symfony's SameOriginCsrfTokenManager
 document.addEventListener('submit', function (event) {
-    var csrfField = event.target.querySelector('input[data-controller="csrf-protection"], input[name="_csrf_token"]');
+    generateCsrfToken(event.target);
+}, true);
```

Comment:
sorry I missed this during the review: why adding true as second argument here @hlecorche ?

Reply:
Doing that in the capture phase means that this listener (attached on the root of the document) runs *before* the listeners attached directly on the form. This ensures that listeners attached on a form to perform custom submission logic (using Ajax for instance) will have the value for the generated CSRF token (instead of generating it too late): https://developer.mozilla.org/en-US/docs/Learn_web_development/Core/Scripting/Event_bubbling#event_capture

Reply:
and this fixes https://github.com/symfony/symfony/issues/59571

Reply:
Here’s an example that illustrates the problem :

```js
document.addEventListener('submit', function (event) {
	if (event.target.matches('form[data-toggle="ajax-form"]')) {
	  event.preventDefault()
	  // Send request with fetch
	}
})
```

Depending on the order of event listener registration, the event that triggers the fetch request can be executed before the csrf_protection_controller.js event listener. This is exactly what happens in my case, as my event listener is registered during the DOMContentLoaded event, which causes it to run before the CSRF protection logic is applied.

To fix this issue, I used the capture phase (true as the third argument in addEventListener) to ensure that the CSRF protection logic is executed before the fetch request is sent.